Gems/Atom/RPI/Code/Source/RPI.Reflect/Image/StreamingImageAsset.cpp

```diff
@@ -51,23 +129,42 @@ namespace AZ
             return 0;
         }
 
+        float ScaleValue(float value, float origMin, float origMax, float scaledMin, float scaledMax)
+        {
+            return ((value - origMin) / (origMax - origMin)) * (scaledMax - scaledMin) + scaledMin;
+        }
+
         float RetrieveFloatValue(const AZ::u8* mem, size_t index, AZ::RHI::Format format)
         {
             switch (format)
             {
             case AZ::RHI::Format::R8_UNORM:
-            case AZ::RHI::Format::R8_SNORM:
             case AZ::RHI::Format::A8_UNORM:
             {
                 return mem[index] / static_cast<float>(std::numeric_limits<AZ::u8>::max());
             }
-            case AZ::RHI::Format::R16_FLOAT:
+            case AZ::RHI::Format::R8_SNORM:
+            {
+                // Scale the value from AZ::s8 min/max to -1 to 1
+                auto actualMem = reinterpret_cast<const AZ::s8*>(mem);
+                return ScaleValue(actualMem[index], std::numeric_limits<AZ::s8>::min(), std::numeric_limits<AZ::s8>::max(), -1, 1);
+            }
```

Comment:
SNORM is tricky. The problem is that min is not the same as negative max, so that makes 0 come out to not-quite 0 using this function. I decided to look up the details of how SNORM is decoded in shaders and it actually treats the minimum two values as equal, so in the case of s8, -128 and -127 would both equal -1.0f. This gives you a symmetric range of -127 to +127, and then 0 really is zero.

Reply:
Oh that makes sense, thank you, I will update